packages/components/src/Menu/Menu.tsx

```diff
@@ -251,29 +260,11 @@ export interface ActionProps {
    * Callback when an action gets clicked
    */
   onClick?(event: React.MouseEvent<HTMLButtonElement>): void;
-
-  /**
-   * Focus on the action when rendered
-   */
-  readonly shouldFocus?: boolean;
 }
 
-function Action({
-  label,
-  sectionLabel,
-  icon,
-  onClick,
-  shouldFocus = false,
-}: ActionProps) {
+function Action({ label, sectionLabel, icon, onClick }: ActionProps) {
   const actionButtonRef = useRef() as RefObject<HTMLButtonElement>;
 
-  useEffect(() => {
-    if (shouldFocus) {
-      // Focus on the next tick to allow useRefocusOnActivator to initialize
-      setTimeout(() => actionButtonRef.current?.focus(), 0);
-    }
-  }, [shouldFocus]);
```

Comment:
we don't need this anymore. at least I'm pretty confident. `useFocusTrap` will both focus the first focusable element, and keep focus trapped

`useFocusTrap` does have a short list of focusable elements, but `button` is one of them, and that's what we're rendering as an `Action`

```
const focusables = [
    "button",
    "[href]",
    "input",
    "select",
    "textarea",
    '[tabindex]:not([tabindex="-1"])',
  ];
```

Reply:
I'm somewhat open to the argument that I didn't need to add `useFocusTrap` for the scope of this ticket. 

I could be convinced, but I will say it feels very bad in practice without it. the focus runs off to the `document.body` because that's where our popper'd menu is now

Reply:
ahh I remember this. `useFocusTrap` and `useRefocusOnActivator` don't play well together

one tries to remember the last `activeElement` assuming it is the original button, and then the other tries to set the activeElement to be the first item in the newly opened dialog

then, when we close it the `useRefocusOnActivator` tries to set the focus back to the `activeElement` which is the now unmounted button inside the dialog 🥴 

fixing the interaction between these two might be fun

@chris-at-jobber if you had to choose one or the other which is preferable?

- refocusing properly when closed, however tabbing out of the Menu will NOT close it, and now that it is on the body you'll be tabbing to stuff miles away from where you were
- tabbing the opened Menu will stay locked in the dialog, but when you close it the focus gets lost and the focus is now on the body 😅 

I will call out that the former is the current experience

Reply:
They work together in Modal don't they? If we have to choose, I would (for now) leave the experience as is with `refocusOnActivator` working and the tabs running out of the menu, and we can revisit this later. We should _actually_ be operating the menu options with arrow keys instead of tabs anyways, which is a separate long-standing issue.

Reply:
oh! that's a very good point. I forgot about Modal

ok scratch that. I'll see what it is doing to make it work.

Reply:
lol, I remember now

you have to call the refocus hook first, before the the focus trap hook or else it won't work

thinking about how we can better signal this temporal requirement beyond my comment. maybe we combine the two into a single hook that always calls them in the right order? trying to think if there's ever a time where you only want one of them and not both

anyways that's something for another day. this should work nicely now!

Reply:
Hey! It works nicely now, just like you said. 🎉